docs/design/auth.md

```diff
@@ -0,0 +1,108 @@
+## Auth with the MCP Gateway (Phase 1)
+
+To provide auth intetgration, the MCP gateway uses (Kuadrant)[https://kuadrant.io] and Gateway API in combination with Keycloak as an identity provider.
+
+We are bringing these components together to provide a reference implementation of the OAuth 2.0 Protected Resource Metadata as per the (MCP Authorization)[https://modelcontextprotocol.io/specification/draft/basic/authorization]  and a reference implementation for authorizing access to particular tool.
+
+> Note while we use keycloak in our examples, in theory other IDPs could also be used.
+
+
+### Authentication Implementation
+
+![](./images/MCPgateway-auth.jpg)
+
+### Protecting the MCP Gateway HTTP endpoint(s)
+
+In order to enforce authenticated access, the gateway listeners exposing the MCP gateway to clients, can use a kuadrant `AuthPolicy` resource. This resource can define what authentication must be met and also enforce Authz requirements at the gateway (Authorization Covered Below).
+
+[Example AuthPolicy](./../../config/mcp-system/authpolicy.yaml)
+
+[Example Oauth Setup](../../README.md#example-oauth-setup)
+
+With this policy in place any unauthenticated request not going to the /.well-known endpoints will get a 401 with a resource_metadata url specified. It is then up to the client to leverage this information to retrieve the resource metadata to know how to authenticate to access the resource (example resource would be /mcp). This resource metadata will be served via the MCP Broker component. 
+
+To understand the full flow including the defined Auth server take a look at [MCP Gateway Auth Sequence Diagram](./flows.md#mcp-gateway-request-authentication)
+
+
+### Registering required scopes
+
+As the MCP gateway is aggregating multiple MCP Servers together. We need to know what scopes are required when asking for an authorization token from the auth provider.  To do this each MCP server registered via the `MCPServer` resource that targets a HTTPRoute at the Gateway will declare its required scopes:
+
+
+```yaml
+
+## not actual implmentation may be slightly different.
+
+name: accounting_mcp
+  spec:
+    # Reference all three test MCP servers via their HTTPRoutes
+    targetRefs:
+    # Server 1 - Go SDK based (hi, time, slow, headers tools)
+    - group: gateway.networking.k8s.io
+      kind: HTTPRoute
+      name: mcp-server1-route
+    toolPrefix: s3_
+    auth: #just an example that may work
+      scopes: 
+        - email
+        - role
+        - user
+      rbac:
+        - role: accounting
+          tools: 
+            - "payroll"
+        - role: admin
+          tools: 
+            -"*"
```

Comment:
If this info can be compiled dynamically from multiple (hypothetical) config files of each tool, then perhaps we want here some kind of selector of the tools/config files themselves instead, as opposed to duplicating the info into the MCPServer resource?

On the other hand, if it's a static list (that occasionally may diverge from the actual tools running/being served behind), then I wonder if the info doesn't belong into the AuthPolicy.

In fact, why having an `auth` section in the `MCPServer` resource et all? Couldn't this (shouldn't this?) all be in the AuthPolicy? Unless we're thinking on hiding the creation/ownership of the AuthPolicy resource behind a MCPServer one as a high level API?

Reply:
So we have 2 options we can show. One the ACL server and 2 via keycloak. I am moving tool filtering to phase 2 doc